### PR TITLE
Add mainnet genesis configuration

### LINKED ISSUE #64: Create genesis block configuration for Bane testnet and mainnet
Just a remind and for discussion, because I haven't seen anything about it in the existing issues.

Now in genesis block, timestamp is '0x0'.
```
{
    "jsonrpc": "2.0",
    "id": 3,
    "result": {
        "baseFeePerGas": "0x3b9aca00",
        "difficulty": "0x1",
        "extraData": "0x0000000000000000000000000000000000000000000000000000000000000000021d70769889c46db0a8b0b22245c7ffd956d343a6b78921e12d527e13a3907da9d6a1fd8e6f8d0abe0c8b1251697f919af32a42b7892b13026433b8d4d71adea0eeb18568fecf0fae0810a8520b438b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "gasLimit": "0x6acfc0",
        "gasUsed": "0x0",
        "hash": "0x98bb61fde1b49ef9e62992a28adfb71df99001a6dec2da1c5d534975f5611263",
        "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
        "miner": "0x0000000000000000000000000000000000000000",
        "mixHash": "0x04b83a5f5b6666f8bba8f78245767c5c1723842a8f442ff0dbe35141eefede71",
        "nonce": "0x0000000000000000",
        "number": "0x0",
        "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "receiptsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
        "sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
        "size": "0x332",
        "stateRoot": "0x7a77851842736c8672a67c64093fcdae5e1991f8cee5806a2d80a4b70a9749a9",
        "timestamp": "0x0",
        "totalDifficulty": "0x1",
        "transactions": [],
        "transactionsRoot": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
        "uncles": []
    }
}
```

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Add genesis configuration for Neo X mainnet. @txhsl, @chenquanyu please review and ensure that contracts code is OK.

Can't be merged until all TODOs are resolved.

Close #64.

### CODE REVIEW

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [inline config/genesis_mainnet.json]: I don't think we need some side balances for mainnet, but maybe I'm wrong. @steven1227, could you confirm?

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [inline config/genesis_mainnet.json]: @steven1227, @gsmachado please, provide StandBy validators addresses for mainnet.

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [inline config/genesis_mainnet.json]: @txhsl, could you tell me where can I get the code of proxy contracts?

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [inline config/genesis_mainnet.json]: @steven1227, @gsmachado we need to decide what is the initial amount of GAS for Treasury contract.

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [inline config/genesis_mainnet.json]: @mialbu what is approximate date when mainnet version of Bridge contracts is finished?

File: config/genesis_mainnet.json
Comment (txhsl) [reply config/genesis_mainnet.json]: Please refer to https://github.com/OpenZeppelin/openzeppelin-contracts/blob/release-v5.0/contracts/proxy/ERC1967/ERC1967Proxy.sol.

This file was added by #7, but we didn't change it, so it was removed in #210.

File: config/genesis_mainnet.json
Comment (txhsl) [reply config/genesis_mainnet.json]: This is the bridge relayer address, if it has no balance to send the first transaction, then other GAS will be locked in bridge and treasury.

Comment (AnnaShaleva) [reply config/genesis_mainnet.json]: Good, reverted. Should we use the same address for mainnet?

Comment (txhsl) [reply config/genesis_mainnet.json]: It depends on the bridge configuration @gsmachado @mialbu.

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json]: I'll create the release PR today. Since I want Shili and Krain to review it, maybe today or tomorrow we have the release and the byte code.

Comment (mialbu) [reply config/genesis_mainnet.json]: Waiting for https://github.com/bane-labs/bridge-evm-contracts/pull/175

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json]: 🚨 This is not the treasury contract. This is the bridge contract. The treasury contract is `0x1212000000000000000000000000000000000006`.

Comment (AnnaShaleva) [reply config/genesis_mainnet.json]: Yes, TODO is placed in the wrong place. But we still need to decide on it.

File: config/genesis_mainnet.json
Comment (mialbu) [inline config/genesis_mainnet.json:104]: @steven1227
For mainnet, are we keeping the following initial values for the Gas bridge? Otherwise, please tell me what values we should use.

Fee for withdrawing: **0.1 GAS**
Minimum withdrawal amount: **1 GAS**
Maximum withdrawal amount: **100'000 GAS**
Max Nr of Deposits per Relay-Tx: **100**

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json]: > But we still need to decide on it.

Yes, of course. Just helping to make sure we don't modify the value at the wrong place here. 😅

File: config/genesis_mainnet.json
Comment (txhsl) [inline config/genesis_mainnet.json]: ~~We need `0x6c6b935b8bbd400000` if we include #275.~~

No, we will have new values for this.

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json]: @steven1227 Can you provide the address for the bridge relayer for mainnet?

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [reply config/genesis_mainnet.json]: Let's wait until #275 is merged and then update config.

Comment (AnnaShaleva) [reply config/genesis_mainnet.json]: @txhsl, the final decision from merged #275 was to reduce register fee from 20000 GAS to 2000 GAS and to use 50% as a service fee. So I suppose we still need to change this value from `0x043c33c1937564800000` to `0x6c6b935b8bbd400000`, right?

File: config/genesis_mainnet.json
Comment (AnnaShaleva) [reply config/genesis_mainnet.json:104]: Is there any decision on this?

File: config/genesis_mainnet.json
Comment (txhsl) [reply config/genesis_mainnet.json]: The latest news is we again reduce from 2000 GAS to 1000 GAS, but keep 50% as the service fee rate, and set the target vote amount to 6,000,000 GAS.

```json
"0x0000000000000000000000000000000000000000000000000000000000000000": "0x01",
"0x0000000000000000000000000000000000000000000000000000000000000001": "0x07", // 7
"0x0000000000000000000000000000000000000000000000000000000000000002": "0x0de0b6b3a7640000", // 1e18
"0x0000000000000000000000000000000000000000000000000000000000000003": "0x04f68ca6d8cd91c6000000", // 6e24
"0x0000000000000000000000000000000000000000000000000000000000000004": "0x3635c9adc5dea00000", // 1e21
"0x0000000000000000000000000000000000000000000000000000000000000005": "0xec40", // 60480
"0x0000000000000000000000000000000000000000000000000000000000000010": "0x07",
"0x0000000000000000000000000000000000000000000000000000000000000011": "0x07",
```

Comment (txhsl) [reply config/genesis_mainnet.json]: Included in #281.

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json:104]: > Is there any decision on this?

@steven1227

Comment (steven1227) [reply config/genesis_mainnet.json:104]: Confirm this

File: config/genesis_mainnet.json
Comment (mialbu) [inline config/genesis_mainnet.json]: These values are not updated for mainnet yet afaik.

@steven1227 FYI

Comment (mialbu) [reply config/genesis_mainnet.json]: @steven1227 Can you provide the roles on the evm side here?

We need the addresses of the following roles:
- owner
- relayer
- validator 1-7
- securityGuard
- governor

Comment (mialbu) [reply config/genesis_mainnet.json]: **Owner** (CommitteeMultiSig)
Storage Key: `0x9016d09d72d40fdae2fd8ceac6b6234c7706214fd39c1cd1e609a0528c199300`
Value: `0x1212000000000000000000000000000000000007`

**Relayer** (NGD)
Storage Key: `0x0000000000000000000000000000000000000000000000000000000000000064`
Value: `0x0aa8095a4438d2006ee6aef706a012b5457c375f`

**Validator 1** (NGD)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94354`
Value: `0x96790618cdd975e18c29af43bc736324d2bfcc27`

**Validator 2** (NGD)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94355`
Value: `0x2c467ed2ec4e8ad1f5a74e884b517814445b563d`

**Validator 3** (lazynode)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94356`
Value: `0x38858eaa6b4424bbf855ec93857622ebdbb520d5`

**Validator 4** (NSPCC)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94357`
Value: `0x0dad0310606e25c5f384dcb49a2f709ff9e540fc`

**Validator 5** (AxLabs)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94358`
Value: `0x351926e55e6976c63a96c6eb84f8c969b475a499`

**Validator 6** (AxLabs)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e94359`
Value: `0x522414d05c1c26b230f265b3b6317928df9b22b5`

**Validator 7** (AxLabs)
Storage Key: `0x46501879b8ca8525e8c2fd519e2fbfcfa2ebea26501294aa02cbfcfb12e9435a`
Value: `0xaedf83da2983a851c38037a3b6a4e368ad0fd794`

**Governor** (AxLabs)
Storage Key: `0x0000000000000000000000000000000000000000000000000000000000000067`
Value: `0xb125b0b6566b2db33eb358e6293b6e08dfdeb87a`

**SecurityGuard** (NGD)
Storage Key: `0x0000000000000000000000000000000000000000000000000000000000000068`
SecurityGuard: `0xbb03c5030cac72e290ae185a8b9b375c58f7a9a6`

Comment (mialbu) [reply config/genesis_mainnet.json]: We'll need the addresses for the following bridge roles:

@steven1227 
- [x] owner -> what's the state of this in regards to https://github.com/bane-labs/bridge-evm-contracts/issues/176?
- [x] relayer
- [x] validator 1
- [x] validator 2
- [x] validator 3
- [x] security guard

@roman-khimov @AnnaShaleva 
- [x] validator 4

Comment (txhsl) [reply config/genesis_mainnet.json]: > owner -> what's the state of this in regards to https://github.com/bane-labs/bridge-evm-contracts/issues/176?

The contract was merged through https://github.com/bane-labs/go-ethereum/pull/279, and its genesis code is in https://github.com/bane-labs/go-ethereum/pull/281.

You can review this modification, and let's set the EVM bridge owner address to `0x1212000000000000000000000000000000000007`.

File: config/genesis_mainnet.json
Comment (chenquanyu) [reply config/genesis_mainnet.json]: > @steven1227, @gsmachado we need to decide what is the initial amount of GAS for Treasury contract.

100 million, same as T4.

File: config/genesis_mainnet.json
Comment (mialbu) [reply config/genesis_mainnet.json]: @AnnaShaleva All role assignments are updated above in my comment.

**Comment by mialbu** [comment]
The allocations for the bridge and bridge management contracts look good to me. 👍🏼

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**
